### ISSUE TITLE
Cannot load generic font family on MacOS

### ISSUE THREAD

**Issue by 38** [MSleepyPanda/rust-font-loader#26]
Hi,

Thanks for the great crate, I used this to load system fonts for my crate Plotters.  Recently we've added a feature that supports generic font families like serif, serif-sans and monospace.

On Linux, query a generic font family will return expected result. But On MacOS, it seems font-loader doesn't have a way to load generic font family and querying `"serif"` will directly return `FontNotFound`.

I believe `CTFontStylisticClass` should be one way to query the generic font class. 

Also there's a behavior difference on MacOS and Linux when font is not found. On Linux it returns the default font, on MacOS it returns an error.

Could you please give some guidance on this stuff ?

Thanks!

**Comment by MSleepyPanda** [comment]
I can't really comment on the MacOS implementation, i don't own a mac device / don't have access. If you time to spare, feel free to do so :)

IIRC some servo engineers provided the original implementation. They since wrote their own implementation, called [`font-kit`](https://docs.rs/font-kit/0.4.0/font_kit/). You should check if it suits your needs!

**Comment by 38** [comment]
Yep, that works correctly on all platforms. But you know I still kind of like this crate because it has way less dependency. (Also for rusttype).

Thanks for the information!
